### PR TITLE
Fix crashes in vm_version and a little further

### PR DESCRIPTION
**Pull request by AntonKozlov**

### CODE REVIEW

**Comment by AntonKozlov** [comment]
Looks very interesting! What is working with the change and what is not?

File: src/hotspot/os_cpu/linux_riscv/atomic_linux_riscv.hpp [hunk 1]
```diff
@@ -130,8 +128,6 @@ template<>
 template<typename I, typename D>
 inline D Atomic::PlatformAdd<8>::add_and_fetch(I add_value, D volatile* dest,
                                                atomic_memory_order order) const {
-  fprintf(stderr, "Atomic::PlatformAdd<8>::add_and_fetch has been reached. Go to %s and check if it works\n", __FILE__);
```
Comment (AntonKozlov) [inline src/hotspot/os_cpu/linux_riscv/atomic_linux_riscv.hpp:133]: It seems like the function now can be seen in action

**Comment by AntonKozlov** [comment]
I have created this review, so I cannot Approve it :)
But I think it can be merged in, if there are no objections.

File: src/hotspot/os_cpu/linux_riscv/atomic_linux_riscv.hpp (hunk 1 above)
Comment (vasyoid) [reply src/hotspot/os_cpu/linux_riscv/atomic_linux_riscv.hpp:133]: Before deleting this line I tested the function by logging its input and return values. So it seems to work correctly

**Pull request merged**